### PR TITLE
Introduce commit

### PR DESCRIPTION
**Pull request by gigbat**

### CODE REVIEW

File: src/main/java/MyThread.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import org.apache.log4j.Logger;
+
+public class MyThread extends Thread {
+    private static final Logger logger = Logger.getLogger(MyThread.class);
+    private Counter counter;
+
+    @Override
+    public void run() {
+        counter = new Counter();
```
Comment (vlykhvar) [inline src/main/java/MyThread.java]: you create two different counters. That's not race :( You need created one in main methods and then increase the count by two threads in the same time.

**Review by vlykhvar — CHANGES_REQUESTED** [review]

File: src/main/java/MyThread.java (hunk 1 above)
Comment (gigbat) [reply src/main/java/MyThread.java]: done

**Review by vlykhvar — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -8,5 +8,69 @@
...
+                    <consoleOutput>true</consoleOutput>
+                    <failsOnError>true</failsOnError>
+                    <linkXRef>false</linkXRef>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-war-plugin</artifactId>
+                <version>3.3.1</version>
+                <configuration>
```
Comment (RomanDubovskyi) [inline pom.xml]: do we need war plugin for this task?

File: pom.xml [hunk 3]
```diff
@@ -8,5 +8,69 @@
...
+                    <version>3.8.0</version>
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
```
Comment (RomanDubovskyi) [inline pom.xml:67]: empty line at the end

File: src/main/java/Counter.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+public class Counter {
+    private int count = 0;
+
+    public int calculate() {
```
Comment (RomanDubovskyi) [inline src/main/java/Counter.java]:
```suggestion
    public int increment() {
```

File: src/main/java/MyRunnable.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+public class MyRunnable implements Runnable {
+    private static final Logger logger = Logger.getLogger(MyRunnable.class);
+    private Counter counter;
+
+    public MyRunnable(Counter counter) {
+        this.counter = counter;
+    }
+
+    public void run() {
+        synchronized (this.counter) {
```
Comment (RomanDubovskyi) [inline src/main/java/MyRunnable.java]: it shoudn't be synchronized, your task is to see how thread race works

File: src/main/java/MyRunnable.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+            for (int i = 0; i <= 100; i++) {
```
Comment (RomanDubovskyi) [inline src/main/java/MyRunnable.java]:
```suggestion
            while (counter.getValue() < 100) {
```

File: src/main/java/MyThread.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+public class MyThread extends Thread {
+    private static final Logger logger = Logger.getLogger(MyThread.class);
+    private Counter counter;
+
+    public MyThread(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
```
Comment (RomanDubovskyi) [inline src/main/java/MyThread.java:12]: same here

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (gigbat) [reply pom.xml]: oops

File: pom.xml (hunk 3 above)
Comment (gigbat) [reply pom.xml:67]: done

File: src/main/java/Counter.java (hunk 4 above)
Comment (gigbat) [reply src/main/java/Counter.java]: done

File: src/main/java/MyRunnable.java (hunk 5 above)
Comment (gigbat) [reply src/main/java/MyRunnable.java]: i think if we use general resources then we will need to use synchronized block

File: src/main/java/MyRunnable.java (hunk 6 above)
Comment (gigbat) [reply src/main/java/MyRunnable.java]: done

File: src/main/java/MyThread.java (hunk 7 above)
Comment (gigbat) [reply src/main/java/MyThread.java:12]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**